src/bitswap/unwant.js

```diff
@@ -1,13 +1,33 @@
 'use strict'
 
 const promisify = require('promisify-es6')
+const CID = require('cids')
 
 module.exports = (send) => {
   return promisify((args, opts, callback) => {
     if (typeof (opts) === 'function') {
       callback = opts
       opts = {}
     }
+
+    // Mirrors block.get's parsing of the CID
+    let cid
+    try {
+      if (CID.isCID(args)) {
+        cid = args
+        args = cid.toBaseEncodedString()
+      } else if (Buffer.isBuffer(args)) {
+        cid = new CID(args)
+        args = cid.toBaseEncodedString()
+      } else if (typeof args === 'string') {
+        cid = new CID(args)
+      } else {
+        return callback(new Error('invalid argument'))
```

Comment:
It's best not to call a callback in a try block. If the callback throws for whatever reason then it'll be caught and then the callback will be called again....which is really hard to debug.

You can just `throw` here, it'll be caught and the callback will be called.